File: Backend/src/main/java/com/bookstore/dto/RegisterRequest.java
package com.bookstore.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class RegisterRequest {
    @NotBlank(message = "Username cannot be empty")
    private String username;

    @NotBlank(message = "Password cannot be empty")
    private String password;

    private String nickname;
}

File: App/lib/src/features/auth/data/models/register_request.dart
class RegisterRequest {
  final String username;
  final String password;
  final String? nickname;

  RegisterRequest({
    required this.username,
    required this.password,
    this.nickname,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
      if (nickname != null) 'nickname': nickname,
    };
  }
}
